authentication/doc.go

```diff
@@ -0,0 +1,54 @@
+// Package authentication provides a client for using the Auth0 Authentication API.
+//
+// # Usage
+//
+//	import (
+//		"github.com/auth0/go-auth0"
+//		"github.com/auth0/go-auth0/authentication"
+//		"github.com/auth0/go-auth0/authentication/database"
+//		"github.com/auth0/go-auth0/authentication/oauth"
+//	)
+//	a, err := authentication.New(
+//		context.Background(),
```

Comment:
Should we use `context.TODO()` (https://github.com/auth0/go-auth0/pull/307)?
Also for other instances in the file.

Reply:
Ah yeah I totally forgot to update the code here when making that change 🙃 Thanks for catching!